src/message_types.rs

```diff
@@ -0,0 +1,249 @@
+            22 => UserStats(Box::new(
+                Self::interpret_message::<mumble::UserStats>(&payload).unwrap(),
+            )),
+            23 => RequestBlob(Self::interpret_message::<mumble::RequestBlob>(&payload).unwrap()),
+            24 => ServerConfig(Self::interpret_message::<mumble::ServerConfig>(&payload).unwrap()),
+            25 => {
+                SuggestConfig(Self::interpret_message::<mumble::SuggestConfig>(&payload).unwrap())
+            }
+            _ => unreachable!(),
+        }
+    }
+}
+
+pub trait HasType {
+    fn get_id(&self) -> u16;
+}
+
+impl HasType for mumble::Version {
+    fn get_id(&self) -> u16 {
+        0
+    }
+}
+
+impl HasType for mumble::UDPTunnel {
+    fn get_id(&self) -> u16 {
+        1
+    }
+}
+
+impl HasType for mumble::Authenticate {
+    fn get_id(&self) -> u16 {
+        2
+    }
+}
+
+impl HasType for mumble::Ping {
+    fn get_id(&self) -> u16 {
+        3
+    }
+}
+
+impl HasType for mumble::Reject {
+    fn get_id(&self) -> u16 {
+        4
+    }
+}
+
+impl HasType for mumble::ServerSync {
+    fn get_id(&self) -> u16 {
+        5
+    }
+}
+
+impl HasType for mumble::ChannelRemove {
+    fn get_id(&self) -> u16 {
+        6
+    }
+}
+
+impl HasType for mumble::ChannelState {
+    fn get_id(&self) -> u16 {
+        7
+    }
+}
+
+impl HasType for mumble::UserRemove {
+    fn get_id(&self) -> u16 {
+        8
+    }
+}
+
+impl HasType for mumble::UserState {
+    fn get_id(&self) -> u16 {
+        9
+    }
+}
+
+impl HasType for mumble::BanList {
+    fn get_id(&self) -> u16 {
+        10
+    }
+}
+
+impl HasType for mumble::TextMessage {
+    fn get_id(&self) -> u16 {
+        11
+    }
+}
+
+impl HasType for mumble::PermissionDenied {
+    fn get_id(&self) -> u16 {
+        12
+    }
+}
+
+impl HasType for mumble::ACL {
+    fn get_id(&self) -> u16 {
+        13
+    }
+}
+
+impl HasType for mumble::QueryUsers {
+    fn get_id(&self) -> u16 {
+        14
+    }
+}
+
+impl HasType for mumble::CryptSetup {
+    fn get_id(&self) -> u16 {
+        15
+    }
+}
+
+impl HasType for mumble::ContextActionModify {
+    fn get_id(&self) -> u16 {
+        16
+    }
+}
+
+impl HasType for mumble::ContextAction {
+    fn get_id(&self) -> u16 {
+        17
+    }
+}
+
+impl HasType for mumble::UserList {
+    fn get_id(&self) -> u16 {
+        18
+    }
+}
+
+impl HasType for mumble::VoiceTarget {
+    fn get_id(&self) -> u16 {
+        19
+    }
+}
+
+impl HasType for mumble::PermissionQuery {
+    fn get_id(&self) -> u16 {
+        20
+    }
+}
+
+impl HasType for mumble::CodecVersion {
+    fn get_id(&self) -> u16 {
+        21
+    }
+}
+
+impl HasType for mumble::UserStats {
+    fn get_id(&self) -> u16 {
+        22
+    }
+}
+
+impl HasType for mumble::RequestBlob {
+    fn get_id(&self) -> u16 {
+        23
+    }
+}
+
+impl HasType for mumble::ServerConfig {
+    fn get_id(&self) -> u16 {
+        24
+    }
+}
+
+impl HasType for mumble::SuggestConfig {
```

Comment:
The content of this file looks hard to maintain. Is there no better way to do this? @svenstaro

Reply:
This doesn't seem like a good solution to me either.